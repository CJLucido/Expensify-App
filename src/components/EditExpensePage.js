//should this still run before the test page is hooked up?

import React from "react";
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense } from "../actions/expenses";
import {removeExpense} from "../actions/expenses";

export class EditExpensePage extends React.Component {
    
    onSubmit= (expense) =>{
        this.props.editExpense(this.props.expense.id , expense);
        this.props.history.push('/');
    }
    
    onClick= () => {
        this.props.removeExpense({id: this.props.expense.id}); 
        this.props.history.push('/');
} //dispatch was removed from these calls to props because it was called in mapDispatchToProps, doi

      render() { return  (
            <div>
            <ExpenseForm
            expense={this.props.expense}
            onSubmit={this.onSubmit}
            />
        
            <button 
             onClick={this.onClick} 
            >Remove</button>
          
            </div>
            ); }

}

const mapToDispatchProps = (dispatch, props) => { //added props here for removeExpense object, using redux docs add own
    return {
        editExpense: (id, expense) => dispatch(editExpense(id, expense)),
        removeExpense: (data) => dispatch(removeExpense(data)) //why use an arbitrary "data" rather than id?
    }
}

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    };
};

export default connect(mapStateToProps, mapToDispatchProps)(EditExpensePage);


//refactor from classless component to class based
//setup mapToDispatchProps editExpense and removeExpense
//pull global editExpense and removeExpense by passing in dispatch and calling those functions to new dispatch prop in mapTo