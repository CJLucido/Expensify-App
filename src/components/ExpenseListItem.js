import React from 'react';
import { Link } from "react-router-dom";
import moment from 'moment';
import numeral from 'numeral';

//for the button in ExpenseListItem.js props.dispatch messes it all up but in ExpenseListFilter.js I can add or remove "props." from before dispatch with no conscequence, why? It is destructured off of props here so you wouldnt call props of props, I dont know why it will work either way in the other, maybe because there is only ever one dispatch?

 const ExpenseListItem = ({ dispatch, id, description, amount, createdAt}) => (  
    <div>
        <Link to={`/poopypooppoop/${id}`}>
        <h3>
        {description}
        </h3>
        </Link>
        <p>
        {numeral(amount / 100).format('$0,0.00')}
         - 
        {moment(createdAt).format('MMMM Do, YYYY')}
        </p>
    </div>
); 


export default ExpenseListItem;

