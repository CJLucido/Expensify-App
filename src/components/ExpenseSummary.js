import React from "react";
import { connect } from "react-redux";
//import ConnectedExpenseList from "./ExpenseList";
import expensesTotal from "../selectors/expenses-total";
import selectExpenses from "../selectors/expenses"
import numeral from 'numeral';

export const ExpenseSummary = ({expenseCount, expensesTotal}) => {

        //const expenseCount = {...expenses}.valueOf();
        const expenseWord = expenseCount === 1 ? "expense" : "expenses";
        const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');

       
            return (
                <div>{
                   <h1> 
                        <p> Viewing {expenseCount} {expenseWord} totalling
                         {formattedExpensesTotal}</p>
                    </h1>
                }</div>
            )
};

//why can't I just call ConnectedExpenseList as it already has state mapped to props

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);

    return {
        expenseCount : visibleExpenses.length,
        expensesTotal: expensesTotal(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpenseSummary)