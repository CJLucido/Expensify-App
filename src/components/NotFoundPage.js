import React from "react";
import {Link} from "react-router-dom";

//anchor tags should be/couldbe used for links that go outside of webpage or app because cant take advantage of clientside rendering anyway
const NotFoundPage = () => (
    <div>404! - <Link to="/">Go Home</Link> 
    </div>
);

export default NotFoundPage;