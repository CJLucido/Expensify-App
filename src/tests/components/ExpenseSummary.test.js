import React from 'react';
import {shallow} from 'enzyme';
import {ExpenseSummary} from "../../components/ExpenseSummary";
//import expenses from "../fixtures/expenses";

test('should render how many expenses and their total', ()=> {
    const wrapper = shallow(<ExpenseSummary 
        expenseCount={2} //dont need to pass in fixtures to the test cases, just pass in values you are testing!!
        expensesTotal={1245624244}
        />)
    expect(wrapper).toMatchSnapshot();
}); 

test('should render 1 expense and its total', ()=> {
    const wrapper = shallow(<ExpenseSummary 
        expenseCount={1}
        expensesTotal={2352244}
        />)
    expect(wrapper).toMatchSnapshot();
});

