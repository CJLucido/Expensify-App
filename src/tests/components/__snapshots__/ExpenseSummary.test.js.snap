// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1 expense and its total 1`] = `
<div>
  <h1>
    <p>
       Viewing 
      1
       
      expense
       totalling
      $23,522.44
    </p>
  </h1>
</div>
`;

exports[`should render how many expenses and their total 1`] = `
<div>
  <h1>
    <p>
       Viewing 
      2
       
      expenses
       totalling
      $12,456,242.44
    </p>
  </h1>
</div>
`;
